pkg/spec/spec.go

```diff
@@ -29,6 +29,12 @@ type Function struct {
 	Spec            FunctionSpec      `json:"spec"`
 }
 
+// ServiceSpec contains service specification for a FunctionSpec
+type ServiceSpec struct {
+	Headless bool `json:"headless"`
```

Comment:
can you add a small inline comment explain what is the purpose of each field? (I am doing the same for the FunctionSpec in other PR)